src/Framework/AspNetCoreAnalyzers/src/CodeFixes/WebApplicationBuilderFixer.cs

```diff
@@ -0,0 +1,165 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace Microsoft.AspNetCore.Analyzers.WebApplicationBuilder.Fixers;
+public class WebApplicationBuilderFixer : CodeFixProvider
+{
+    public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(new[]
+    {
+        // Add other diagnostic descriptor id's
+        DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id
+     });
+
+    public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+    public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+    {
+        foreach (var diagnostic in context.Diagnostics)
+        {
+            context.RegisterCodeFix(
+                CodeAction.Create("Fix Configure methods",
+                    cancellationToken => FixConfigureMethods(diagnostic, context.Document, cancellationToken),
+                    equivalenceKey: DiagnosticDescriptors.DisallowConfigureAppConfigureHostBuilder.Id),
+                diagnostic);
+        }
+
+        return Task.CompletedTask;
+    }
+
+    private static async Task<Document> FixConfigureMethods(Diagnostic diagnostic, Document document, CancellationToken cancellationToken)
+    {
+        // hey Safia some of the below are just draft comments to help you understand the code easier.
+        // We're gonna delete it later before the final commit
+        
+        DocumentEditor editor = await DocumentEditor.CreateAsync(document, cancellationToken);
+        var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+        if (root == null)
+        {
+            return document;
+        }
+
+        var invocation = root.FindNode(diagnostic.Location.SourceSpan); //whole invocation starting from "builder"
+      
+        if (invocation is InvocationExpressionSyntax { } newInvocationExpr)
+        {
+            if (newInvocationExpr.Expression is not MemberAccessExpressionSyntax hostBasedInvocationMethodExpr)
+            {
+                return editor.OriginalDocument;
+            }
+
+            var configuration = SyntaxFactory.IdentifierName("Configuration");
+
+            if (hostBasedInvocationMethodExpr.Expression is not MemberAccessExpressionSyntax subExpression)
+            {
+                return editor.OriginalDocument;
+            }
+            if (subExpression.Name != null) //subExpression.Name is Host/Webhost
```

Comment:
Is `Name` actually nullable here? Also, it might help to invert the if statement and return early here? I'm having trouble visualizing what happens if we _don't_ hit this case.